apollo-router/src/json_ext.rs

```diff
@@ -26,10 +26,20 @@ pub(crate) type Object = Map<ByteString, Value>;
 const FRAGMENT_PREFIX: &str = "... on ";
 
 static TYPE_CONDITIONS_REGEX: Lazy<Regex> = Lazy::new(|| {
-    Regex::new(r"(?:\|\[)(?<condition>.+?)(?:,\s*|)(?:\])")
+    Regex::new(r"(?:\|\[(?<condition>.+)?\])")
```

Comment:
You probably still want the `.+` to be lazy instead of greedy, so that you capture the first `]` instead of the last one. (This is really just to accommodate future potential syntax changes, and isn't strictly necessary now for correctness.)
```suggestion
    Regex::new(r"(?:\|\[(?<condition>.+?)?\])")
```